### PR TITLE
ZK-4346: datebox constraint parsing errors

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zul/src/org/zkoss/zul/Datebox.java [hunk 1]
```diff
@@ -862,6 +862,8 @@ public boolean shallBlock(AuRequest request) {
 	 */
 	// -- super --//
 	public void setConstraint(String constr) {
+		if (!isValidConstraint(constr))
```
Comment (DevChu) [inline zul/src/org/zkoss/zul/Datebox.java]: should put in SimpleDateConstraint

Comment (DevChu) [reply zul/src/org/zkoss/zul/Datebox.java]: should comment that using custom DateConstraint is recommanded

File: zul/src/org/zkoss/zul/SimpleDateConstraint.java [hunk 2]
```diff
@@ -128,9 +128,26 @@ public SimpleDateConstraint(int flags, Date begin, Date end, String errmsg) {
...
 		super(constraint);
+		if (!isValidConstraint(constraint))
+			throw new UiException("Invalid constraint: " + constraint + " (the constraint should be formatted in yyyyMMdd)");
 		fixConstraint();
 	}
 
+	private boolean isValidConstraint(String constraint) {
+		int constraintFormatLength = "yyyyMMdd".length(); // default constraint format is yyyyMMdd
+		String[] arr = constraint.replaceAll(",", " ").split("\\s+");
+		for (String each : arr) {
```
Comment (plijyfes) [inline zul/src/org/zkoss/zul/SimpleDateConstraint.java]: use regexp

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**